packages/kotti-ui/source/kotti-field-currency/KtFieldCurrency.vue

```diff
@@ -0,0 +1,235 @@
+<template>
+	<KtField
+		:field="modifiedField"
+		:getEmptyValue="() => null"
+		:helpTextSlot="$slots.helpText"
+	>
+		<input
+			ref="inputRef"
+			v-bind="inputProps"
+			@input="onInput($event.target.value)"
+		/>
+	</KtField>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import {
+	defineComponent,
+	computed,
+	ref,
+	watch,
+	UnwrapRef,
+} from '@vue/composition-api'
+
+import { KtField } from '../kotti-field'
+import { useField, useForceUpdate } from '../kotti-field/hooks'
+import { useI18nContext } from '../kotti-i18n/hooks'
+import { KottiI18n } from '../kotti-i18n/types'
+import { makeProps } from '../make-props'
+import { DecimalSeparator } from '../types/kotti'
+import { isNumberInRange } from '../utilities'
+
+import { KOTTI_FIELD_CURRENCY_SUPPORTS, VALID_REGEX } from './constants'
+import { parseCurrencyUserInput } from './input-parser'
+import { KottiFieldCurrency } from './types'
+import {
+	toFixedPrecisionString,
+	toNumber,
+	replaceDecimalSeparator,
+	truncate,
+} from './utilities'
+
+const shouldClear = (newValue: string, oldValue: string | null) => {
+	const isClearingZero = !toNumber(newValue) && !toNumber(oldValue)
+
+	return isClearingZero && newValue.length < oldValue?.length
+}
+
+export default defineComponent<KottiFieldCurrency.PropsInternal>({
+	name: 'KtFieldCurrency',
+	components: { KtField },
+	props: makeProps(KottiFieldCurrency.propsSchema),
+	setup(props, { emit, root }) {
+		const field = useField<KottiFieldCurrency.Value>({
+			emit,
+			isCorrectDataType: (value): value is KottiFieldCurrency.Value =>
```

Comment:
This may be unnecessary after the zod fields PR

Reply:
separate PR is better after we merge this, to affect all fields